test/unit/calculators/checks_benefits_support_calculator_test.rb

```diff
@@ -221,473 +335,709 @@ class CheckBenefitsSupportCalculatorTest < ActiveSupport::TestCase
+            calculator.where_do_you_live = "northern-ireland"
+            calculator.over_state_pension_age = "yes"
+            assert calculator.eligible_for_housing_benefit_northern_ireland?
+          end
         end
 
-        should "return false if not eligible for Housing Benefit (Northern Ireland)" do
-          calculator = CheckBenefitsSupportCalculator.new
-          %w[england wales scotland].each do |country|
-            calculator.where_do_you_live = country
+        context "when ineligible" do
+          should "be false if country is not Northern Ireland" do
+            %w[england wales scotland].each do |country|
+              calculator = CheckBenefitsSupportCalculator.new
+              calculator.where_do_you_live = country
+              calculator.over_state_pension_age = "yes"
+              assert_not calculator.eligible_for_housing_benefit_northern_ireland?
+            end
+          end
+
+          should "be false if country is Northern Ireland and under state pension age" do
+            calculator = CheckBenefitsSupportCalculator.new
+            calculator.where_do_you_live = "northern-ireland"
             calculator.over_state_pension_age = "no"
             assert_not calculator.eligible_for_housing_benefit_northern_ireland?
           end
         end
       end
 
       context "#eligible_for_tax_free_childcare?" do
-        should "return true if eligible for Tax Free Childcare without a disabled child" do
-          calculator = CheckBenefitsSupportCalculator.new
-          calculator.are_you_working = "yes_over_16_hours_per_week"
-          calculator.children_living_with_you = "yes"
-          %w[1_or_under 2 3_to_4 5_to_11].each do |age|
-            calculator.age_of_children = age
-            assert calculator.eligible_for_tax_free_childcare?
+        context "when eligible" do
+          should "be true if working over 16 hours, with children between 1 and 11" do
+            calculator = CheckBenefitsSupportCalculator.new
+            calculator.are_you_working = "yes_over_16_hours_per_week"
+            calculator.children_living_with_you = "yes"
+            %w[1_or_under 2 3_to_4 5_to_11].each do |age|
+              calculator.age_of_children = age
+              assert calculator.eligible_for_tax_free_childcare?
+            end
           end
-        end
 
-        should "return true if eligible for Tax Free Childcare with a disabled child" do
-          calculator = CheckBenefitsSupportCalculator.new
-          calculator.are_you_working = "yes_over_16_hours_per_week"
-          calculator.children_living_with_you = "yes"
-          calculator.children_with_disability = "yes"
-          %w[1_or_under 2 3_to_4 5_to_11 12_to_15 16_to_17].each do |age|
-            calculator.age_of_children = age
-            assert calculator.eligible_for_tax_free_childcare?
+          should "be true if working over 16 hours with a disabled child and children between 1 and 17" do
```

Comment:
maybe change to 'be true if working, with a disabled child' as can be working more or less than 16 hours per week and still be eligible